server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -477,10 +473,10 @@ public String toString() {
         }
         sb.append(blocks());
         sb.append(nodes());
-        if (transportVersions.isEmpty() == false) {
+        if (versionsWrappers.isEmpty() == false) {
             sb.append("transport versions:\n");
-            for (var tv : transportVersions.entrySet()) {
-                sb.append(TAB).append(tv.getKey()).append(": ").append(tv.getValue()).append("\n");
+            for (var tv : versionsWrappers.entrySet()) {
+                sb.append(TAB).append(tv.getKey()).append(": ").append(tv.getValue().transportVersion()).append("\n");
```

Comment:
Suggest exposing the whole `VersionsWrapper` here.

Reply:
The irony of writing this after "I can't really think of a good situation for exposing the whole `VersionsWrapper`" in my previous comment is not lost on me 😆